cmp/grid/Grid.ts

```diff
@@ -472,65 +474,53 @@ export class GridLocalModel extends HoistModel {
+                        agColState.indexOf(autoColState),
+                        0,
+                        autoColState as ColumnState
+                    );
                 }
 
-                // 1) Columns all in right place -- simply update incorrect props we maintain
-                if (
-                    isEqual(
-                        colState.map(c => c.colId),
-                        agColState.map(c => c.colId)
-                    )
-                ) {
-                    let hasChanges = false;
-                    colState.forEach((col, index) => {
-                        const agCol = agColState[index],
-                            id = col.colId;
-
-                        if (agCol.width !== col.width) {
-                            api.setColumnWidths([{key: id, newWidth: col.width}]);
+                // Determine if column order has changed
+                const applyOrder = !isEqual(
+                    colState.map(c => c.colId),
+                    agColState.map(c => c.colId)
+                );
+
+                // Build a list of column state changes
+                colState = compact(
+                    colState.map(({colId, width, hidden, pinned}) => {
+                        const agCol: AgColumnState = agColState.find(c => c.colId === colId) || {
+                                colId
+                            },
+                            ret: any = {colId};
+
+                        let hasChanges = applyOrder;
+
+                        if (agCol.width !== width) {
+                            ret.width = width;
                             hasChanges = true;
                         }
-                        if (agCol.hide !== col.hidden) {
-                            api.setColumnsVisible([id], !col.hidden);
+
+                        if (agCol.hide !== hidden) {
+                            ret.hide = hidden;
                             hasChanges = true;
                         }
-                        if (agCol.pinned !== col.pinned) {
-                            api.setColumnsPinned([id], col.pinned);
+
+                        if (agCol.pinned !== pinned) {
+                            ret.pinned = pinned;
                             hasChanges = true;
                         }
-                    });
-
-                    // We need to tell agGrid to refresh its flexed column sizes due to
-                    // a regression introduced in 25.1.0.  See #2341
-                    if (hasChanges) {
-                        api.columnModel.refreshFlexedColumns({
-                            updateBodyWidths: true,
-                            fireResizedEvent: true
-                        });
-                    }
-
-                    return;
-                }
 
-                // 2) Otherwise do an (expensive) full refresh of column state
-                // Merge our state onto the ag column state to get any state which we do not yet support
```

Comment:
So this key change here - where we are now only going to pass any _different_ state, relies on the partial update behavior [documented here](https://www.ag-grid.com/javascript-data-grid/column-state/#applying-partial-state), is that correct?

I went looking for the changelog entry just to confirm when that change happened on the AG Grid side - I think I found it, but it's listed as fixed in 32.2.0 and we are on 31.x:

<img width="1358" alt="Screenshot 2025-01-15 at 6 33 15 PM" src="https://github.com/user-attachments/assets/8d344578-9423-4a27-9abd-681cfdb1f336" />

Does this change need us to take that update, or am I missing something else here?

Reply:
Hmm they certainly claimed that it was supported in 31.x in the way we are using it, and it seemed to work fine in my testing. Not clear what that feature request actually changed, doesn't looks like the docs for that were changed at all.

 https://www.ag-grid.com/archive/31.3.2/javascript-data-grid/column-state/#applying-partial-state

Reply:
Support ticket filed with AG for clarification, just to be careful here (although as discussed, good for us to get to 32.x soon anyway)

Reply:
Confirmed by AG that ticket related to a new flag `partialColumnState` that relates to _initial_ grid state only - https://www.ag-grid.com/react-data-grid/grid-state/#reference-GridState-partialColumnState